examples/AnalyticsTelemetrics/ViewModels/TelemetryPageViewModel.cs

```diff
@@ -0,0 +1,150 @@
+﻿using AnalyticsTelemetrics.Models;
+using AnalyticsTelemetrics.Services;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+
+namespace AnalyticsTelemetrics.ViewModels
+{
+    public partial class TelemetryPageViewModel : ObservableObject
+    {
+        // Configuration values for the sensor collection
+        private const int _numberOfSensors = 5;
+        private const int _collectionDelayMilliseconds = 500;
+
+        private readonly List<Sensor> _sensors;
+
+        private CancellationTokenSource _cancellationTokenSource;
+        private Task _sensorCollectionTask;
+
+        [ObservableProperty]
+        private string _logText = string.Empty;
+
+        [ObservableProperty]
+        [NotifyCanExecuteChangedFor(nameof(StopSensorCollectionCommand))]
+        [NotifyCanExecuteChangedFor(nameof(StartSensorCollectionCommand))]
+        private bool _isCollectionRunning = false;
+
+        public TelemetryPageViewModel()
+        {
+            _sensors = new List<Sensor>();
+            for (int i = 1; i <= _numberOfSensors; i++)
+            {
+                _sensors.Add(new Sensor(i));
+            }
+        }
+
+        [RelayCommand]
+        public async Task OnDisappearing()
+        {
+            await StopSensorCollection();
+        }
+
+        [RelayCommand(CanExecute = nameof(CanStartCollection))]
+        public void StartSensorCollection()
+        {
+            if (IsCollectionRunning)
+            {
+                return;
+            }
+
+            _cancellationTokenSource = new CancellationTokenSource();
+
+            _sensorCollectionTask = Task.Run(() => SensorCollection(_cancellationTokenSource.Token));
+            IsCollectionRunning = true;
+        }
+
+        [RelayCommand(CanExecute = nameof(CanStopCollection))]
+        public async Task StopSensorCollection()
+        {
+            if (!IsCollectionRunning)
+            {
+                return;
+            }
+
+            _cancellationTokenSource.Cancel();
+            await _sensorCollectionTask;
+
+            _cancellationTokenSource = null;
+            _sensorCollectionTask = null;
+            IsCollectionRunning = false;
+        }
+
+        private bool CanStopCollection() => IsCollectionRunning;
+
+        private bool CanStartCollection() => !IsCollectionRunning;
+
+        private async Task SensorCollection(CancellationToken cancellationToken)
+        {
+            AddToLog("Collection started");
+
+            try
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    foreach (var sensor in _sensors)
+                    {
+                        var realm = RealmService.GetRealm();
```

Comment:
I think you saw it down here, but it's because I'm awaiting asynchronously on Task.Delay, so we could switch thread after that. But I'll change that part so we don't need this. And yes, I'll add the using